mono/metadata/class-init.c

```diff
@@ -3846,6 +3847,35 @@ mono_class_layout_fields (MonoClass *klass, int base_instance_size, int packing_
 			real_size = MAX (real_size, size + field_offsets [i]);
 		}
 
+		/* check for incorrectly aligned or overlapped by a non-object field */
+		layout_check = g_new0 (guint8, real_size);
+		if (layout == TYPE_ATTRIBUTE_EXPLICIT_LAYOUT) {
+			for (i = 0; i < top && !mono_class_has_failure (klass); i++) {
```

Comment:
This is already in the EXPLICIT_LAYOUT case, so the if is not needed. Instead, it can use klass->has_references.
Also, there is already a loop below which seems to do the same thing ?

Reply:
I did the new loop because I need the real_size to create the layout_check array...

Reply:
I mean the loop right after this loop seems to do the exact same thing.

Reply:
Thanks!!!